superset/security/manager.py

```diff
@@ -2673,12 +2666,9 @@ def raise_for_ownership(self, resource: Model) -> None:
         :raises SupersetSecurityException: If the current user is not an owner
         """
 
-        # pylint: disable=import-outside-toplevel
-        from superset import db
-
         if self.is_admin():
             return
-        orig_resource = db.session.query(resource.__class__).get(resource.id)
+        orig_resource = self.get_session.query(resource.__class__).get(resource.id)
```

Comment:
We should use the Flask-AppBuilder session everywhere for consistency. Note that `db.session` and `self.get_session` are the same Flask-SQLAlchemy session so the logic remains unchanged.